scripts/launch_guest.sh

```diff
@@ -71,6 +72,22 @@ while [[ $# -gt 0 ]]; do
       shift
       shift
       ;;
+    --snapshot)
+      case $2 in
+        on)
+          SNAPSHOT="on"
+          ;;
+        off)
+          SNAPSHOT="off"
+          ;;
+        *)
+          echo "ERROR: Invalid option for --snapshot: $2" >&2
+          exit 1
+          ;;
+      esac
```

Comment:
What about removing the parsing here, since QEMU will already do that.

I mean this:
```suggestion
      SNAPSHOT="$2"
```

With that applied, QEMU will print:
```
$ ./scripts/launch_guest.sh --image /tmp/image --snapshot INVALID
...

qemu-system-x86_64: -drive file=/tmp/image,if=none,id=disk0,format=qcow2,snapshot=INVALID: Parameter 'snapshot' expects 'on' or 'off'
```